undertow-servlet/src/main/java/org/wildfly/elytron/web/undertow/server/servlet/ElytronHttpServletExchange.java

```diff
@@ -0,0 +1,304 @@
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+import javax.servlet.http.HttpSession;
+
+import org.jboss.logging.Logger;
+import org.wildfly.elytron.web.undertow.server.ElytronHttpExchange;
+import org.wildfly.elytron.web.undertow.server.ScopeSessionListener;
+import org.wildfly.security.http.HttpScope;
+import org.wildfly.security.http.HttpScopeNotification;
+import org.wildfly.security.http.Scope;
+
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.session.SessionConfig;
+import io.undertow.server.session.SessionManager;
+import io.undertow.servlet.api.Deployment;
+import io.undertow.servlet.handlers.ServletRequestContext;
+import io.undertow.servlet.util.SavedRequest;
+
+/**
+ * An extension of {@link ElytronHttpExchange} which adds servlet container specific integrations.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ */
+class ElytronHttpServletExchange extends ElytronHttpExchange {
+
+    private static final Logger log = Logger.getLogger("org.wildfly.security.http.servlet");
+
+    private final HttpServerExchange httpServerExchange;
+    private final ScopeSessionListener scopeSessionListener;
+
+    static Function<HttpServerExchange, HttpScope> APPLICATION_SCOPE_RESOLVER = ElytronHttpServletExchange::applicationScope;
+
+    protected ElytronHttpServletExchange(final HttpServerExchange httpServerExchange, final ScopeSessionListener scopeSessionListener) {
+        super(httpServerExchange);
+        this.httpServerExchange = httpServerExchange;
+        this.scopeSessionListener = scopeSessionListener;
+    }
+
+    @Override
+    protected SessionManager getSessionManager() {
+        ServletRequestContext servletRequestContext = httpServerExchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY);
+        return servletRequestContext.getDeployment().getSessionManager();
+    }
+
+    @Override
+    protected SessionConfig getSessionConfig() {
+        ServletRequestContext servletRequestContext = httpServerExchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY);
+        return servletRequestContext.getCurrentServletContext().getSessionConfig();
+    }
+
+    @Override
+    public int forward(String path) {
+        final ServletRequestContext servletRequestContext = httpServerExchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY);
+        ServletRequest req = servletRequestContext.getServletRequest();
+        ServletResponse resp = servletRequestContext.getServletResponse();
+        RequestDispatcher disp = req.getRequestDispatcher(path);
+
+        final FormResponseWrapper respWrapper = httpServerExchange.getStatusCode() != OK && resp instanceof HttpServletResponse
+                ? new FormResponseWrapper((HttpServletResponse) resp) : null;
+
+        try {
+            disp.forward(req, respWrapper != null ? respWrapper : resp);
```

Comment:
Coverity is complaining 

https://u2389337.ct.sendgrid.net/wf/click?upn=08onrYu34A-2BWcWUl-2F-2BfV0V05UPxvVjWch-2Bd2MGckcRYZquIFszCcwQkbx6rX2y-2BUxNCGJdkIRvpY0lX6FsmGlUNLm1ZLqjmn3Xwt68CpLFqJgqd9fwu9t55-2Fv2R1w01q_2fPkMU1miRf-2FtjnMal9VVM35TqEfZozw4WTtk2YIMAKSiRrskMGOQZz0Ta8rCKNvNs3WUbHVZvDmaDHeD309T5r6-2BsMVqQB-2BO6BuefCCb4s1a50ellRFmuTIofeoY143RBQ8RnqVwL-2F1meA8j0sJepALas4QiG2qruipA-2B4k4D6BX9I62R4IWYXHSEAOTESI8QP9dueBxyBpGXjdQ3yhqw-3D-3D

Reply:
https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getRequestDispatcher(java.lang.String)

Reply:
Do you have details of the error?  To click that link I need to grant loads of permissions just to see it.

Reply:
Javadoc for ServletRequest.getRequestDispatcher(path) says null can be returned. Therefore Coverity detect disp.forward() as potentially dereferencing null code.

Reply:
I think if we can get an issue created here we can take a look https://issues.jboss.org/projects/ELYWEB/summary the real objective of this step was to pull in the code as-is as much as possible to form the baseline of the porting - we currently don't have any specific error messages for elytron web yet but we will likely need them for this.

Reply:
I have created https://issues.jboss.org/browse/ELYWEB-6